### ISSUE TITLE
Serialization Upgrade

### ISSUE THREAD

**Issue by charlie-map** [charlie-map/CS_158C#7]
The current serialization algorithm follows directly that of a book where it contains words and which pages they occur on:

```
cat: 15, 32
cow: 29, 30, 45
```

However, since there are multiple books (that contain many of the same words), the index starts to look like:

```
1 cat:15,32 cow:29,30,45
2 cat:45,60 dog:10,23
```

When the number of documents starts reaching many thousands, the word `cat` will be stored thousands of times. However, with this update version (see below for more), the occurrences of each word would be a constant of 1, and in exchange, the document (or book) identification would occur for each word within the document. Since document identification is (on average) significantly shorter than words, this will likely prove advantageous. The following is an example of what the two previous books would like with this new form proposal:

```
cat:1,2|2,2
cow:1,3
dog:2,2
```

where the digit immediately after a `:` or a `|` is the ID number and the number after the `,` is the occurrence number within that said document. To inform more understanding moving forward, this issue will act as a journal for this process, including comparisons in efficiency between these two strategies.

**Comment by charlie-map** [comment]
Because of the utilization of this, the necessity of having an entire hashmap just for document frequency is now obsolete because the document frequency can be stored right along next to the term frequencies.

**Comment by charlie-map** [comment]
With the fix from #9, the serialization update is officially complete.
